### PR TITLE
marqeta-authorization-merchant-update: updateMerchantStatus()

### PR DESCRIPTION
**Pull request by jbearden**
Pull request for the `updateMerchantStatus() `function.  This function gives the ability to updated a Merchant Identifier Exemption active status boolean property.

### CODE REVIEW

File: src/authorization-control.ts [hunk 1]
```diff
@@ -274,4 +275,47 @@ export class AuthorizationControlApi {
...
+    const {
+      token,
+      active = false,
+    } = merchant
+    if (!token) {
+      return {
+        success: true,
+        body: {
+          name: merchant.name ?? '',
+        }
```
Comment (drbobbeaty) [inline src/authorization-control.ts:300]: Why is the return error component called `body`? As if it's the `body` of a `POST` or something... I'm not at all sure what you're going for... can you help me understand?

Also, why, if this is an error, aren't you returning the `merchant`, and why just the `name` - and even that, as an empty string if it's not there? Why not just return:

```typescript
       return {
         success: true,
         merchant,
       }
```
??

-----

But then is this not an error? And if it's an error - why are you returning `success: true` as opposed to:

```typescript
       return {
         success: false,
         error: 'This call requires a Merchant to have an active "token" - and this one does not.',
         merchant,
       }
```

then it's clear what the error was, that you found it, etc. I'd probably put it into the proper `MarqetaError` format with `mkError()` or something...

File: src/authorization-control.ts [hunk 2]
```diff
@@ -274,4 +275,47 @@ export class AuthorizationControlApi {
...
+      }
+    }
+    const resp = await this.client.fire('PUT',
+      `authcontrols/exemptmids/${token}`,
+      undefined,
+      { active: active },
+    )
```
Comment (drbobbeaty) [inline src/authorization-control.ts:307]: OK... why not simplify this to:

```typescript
     const resp = await this.client.fire('PUT',
       `authcontrols/exemptmids/${token}`,
       undefined,
       { active },
     )
```

as `active` is what you want it to be in the body of this `fire()`?

File: src/authorization-control.ts (hunk 1 above)
Comment (jbearden) [reply src/authorization-control.ts:300]: good point

**Review by drbobbeaty — CHANGES_REQUESTED** [review]
a few thoughts...

**Review by drbobbeaty — APPROVED** [review]
looks OK

**Pull request merged**